modules/memfault-firmware-sdk/Kconfig

```diff
@@ -187,6 +187,7 @@ config MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP
 
 config MEMFAULT_CDR_ENABLE
 	bool "Custom Data Recording"
+	default y if NRF_MODEM_LIB_TRACE_SHELL
```

Comment:
Is it a good practice to refer to a Kconfig of a client of a module given the client is elsewhere in the code tree?
I guess we could handle this within nrf_modem_lib Kconfig file?

Reply:
Or could it be that libraries are not able to override a Kconfig. Not sure...

What about using `select MEMFAULT_CDR_ENABLE` (could be `imply` or `depends on` too) in nrf_modem_lib Kconfig file?

> This kconfig is one we made in sdk-nrf for our wrapper around the memfault lib, so I don't see why that is problematic.

Maybe it's just me that sees these references from a library Kconfig to its client as suboptimal from architectural point of view. You can do that as Kconfig allows references anywhere but I just feel it's a bit odd. In my opinion it should be the client that defines the libraries it's using.

Reply:
I've tried the different alternatives, but `select MEMFAULT_CDR_ENABLE if MEMFAULT` seemed to work the best.
- `imply` will cause a compilation error without more `#ifdef`s in the code, and `MEMFAULT_CDR_ENABLE` would also still end up as `y` even with `CONFIG_MEMFAULT_CDR_ENABLE=n` in a config file.
- `depends on` can't have an `if`-expression